cli/tests/test_observability_v8_writer.py

```
@@ -207,6 +207,74 @@ def validator(_candidate: str, _data_dir: str | None) -> None:
     assert "retention_days: 45" in path.read_text()
 
 
+def test_mutate_v8_config_rejects_stale_preview_digest(tmp_path: Path) -> None:
+    path = tmp_path / "config.yaml"
+    path.write_text(_source())
+    backup = tmp_path / "backups" / "config.yaml.before-redaction"
+
+    with pytest.raises(RuntimeError, match="changed after.*preview"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)